src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -80,7 +83,7 @@ public static string GetFullPath(string path, string basePath)
                 // Path is current drive rooted i.e. starts with \:
                 // "\Foo" and "C:\Bar" => "C:\Foo"
                 // "\Foo" and "\\?\C:\Bar" => "\\?\C:\Foo"
-                combinedPath = Join(GetPathRoot(basePath.AsSpan()), path);
+                combinedPath = Join(GetPathRoot(basePath.AsSpan()), path.AsSpan().Slice(1)); // Cut the separator to ensure we don't end up with two separators when joining with the root.
```

Comment:
>path.AsSpan().Slice(1) [](start = 68, length = 22)

It feels like Join should be able to handle that, but per comment it is not... after the emergencial fix is done this should be evaluated.